pynio/instance.py

```diff
@@ -91,8 +91,11 @@ def _get_blocks(self):
         blocks = {}
         for bname, config in self._get('blocks').items():
             btype = config['type']
-            b = blocks_types[btype].copy(bname, instance=self)
-            b.config = config
+            b = deepcopy(blocks_types[btype])
+            b._name = bname
+            b._instance = self
+            b._config = config
```

Comment:
I'm thinking this is wrong anyway. it should be b.config = config. But if we change to use add_block then everything will work fine.